### PR TITLE
#319: Increase Backend Code Coverage

### PR DESCRIPTION
**Pull request by DPigeon**
# Related Issue
- #319 

# Proposed changes
- Refactored some classes and tests (mostly tests)
- Added many missing unit and integration tests (about 40 in total)

# Additional Info
- **I left 2 TODOs since this story ended up being very big. I'll stop here and someone (or myself) can complete the rest of the missing tests later. Note that more than 80% of the missing tests were added.**

# Checklist (if applicable)
- [x] Tests

# Reviewer(s)
 - @razine-bensari 
 - @EspressoCode 
 - @abdulmansour 
 - @wltrf

### CODE REVIEW

**Comment by EspressoCode** [comment]
Wow this is absolutely amazing 👏 good job. There's one code smell which is in one of the files I added before, not sure how I missed it, please resolve it here.

File: backend/src/test/groovy/com/capstone/moneytree/controller/UserControllerTest.groovy [hunk 1]
```diff
@@ -598,12 +592,11 @@ class UserControllerTest extends Specification {
         def response = userController.unfollowUser(user1.getId(), user2.getId())
 
         then: "user1 should have unfollowed user2"
-        //        assert user1.getFollowers().isEmpty()
+        // assert user1.getFollowers().isEmpty()
```
Comment (EspressoCode) [inline backend/src/test/groovy/com/capstone/moneytree/controller/UserControllerTest.groovy:595]: Is this not working?

**Review by EspressoCode — APPROVED** [review]

File: backend/src/main/java/com/capstone/moneytree/facade/MarketInteractionsFacade.java [hunk 2]
```diff
@@ -224,10 +215,6 @@ private AlpacaStreamListener createStreamListener(String userId, SimpMessagingTe
       return new AlpacaStreamListenerAdapter(messageType) {
          @Override
          public void onStreamUpdate(AlpacaStreamMessageType streamMessageType, AlpacaStreamMessage streamMessage) {
-            // if (streamMessageType == AlpacaStreamMessageType.TRADE_UPDATES) {
```
[the remark this answers is not in the document]
Comment (DPigeon) [reply backend/src/main/java/com/capstone/moneytree/facade/MarketInteractionsFacade.java:227]: Yeah I removed some unused code.

File: backend/src/test/groovy/com/capstone/moneytree/controller/UserControllerTest.groovy (hunk 1 above)
Comment (DPigeon) [reply backend/src/test/groovy/com/capstone/moneytree/controller/UserControllerTest.groovy:595]: This line was already commented out. I just removed the extra space.

**Comment by DPigeon** [comment]
> Wow this is absolutely amazing 👏 good job. There's one code smell which is in one of the files I added before, not sure how I missed it, please resolve it here.

Thanks! I just pushed a fix for that code smell.

**Pull request merged**